### PR TITLE
Update to match new attributes specification.

### PR DESCRIPTION
**Pull request by barelyknown**
Addresses [this issue](https://github.com/eneuhauser/ember-json-api/issues/10).

### CODE REVIEW

**Comment by fotinakis** [comment]
:+1:

**Comment by doctorloaf** [comment]
yes please

File: tests/integration/specs/updating-an-individual-resource-test.js [hunk 1]
```diff
@@ -108,8 +118,10 @@ asyncTest("Update a post with an author", function() {
   var request = {
     data: {
       id: '1',
-      title: 'TDD Is Dead lol',
-      'post-summary': null,
+      attributes: {
+        title: 'TDD Is Dead lol',
```
Comment (doctorloaf) [inline tests/integration/specs/updating-an-individual-resource-test.js:122]: :joy:

**Comment by wzowee** [comment]
Need to also include key === 'id' in the serialize function when checking keys to skip otherwise id is being pushed into the attributes hash which is not where it's expected as per the current spec.

```
  if (key === 'links' || key === 'attributes' || key === 'id') {
    continue;
  }
```

**Comment by barelyknown** [comment]
Ah. Good catch. I'll update.

**Comment by barelyknown** [comment]
This change actually isn't needed. I added a test to validate. The `serialize` function in the `RESTAdapter` does not include the `id` be default.

**Comment by wzowee** [comment]
The tests aren't calling the serialize function in the same manner as it's being used in the JSONAdapter and I think this is the problem.

JSONAdapter

```
data: serializer[fn](snapshot, { includeId:true, type:type.typeKey })
```

serializer-test

```
env.serializer.serialize(snapshot)
```

If you update the serializer-test to the following it will fail with id being included in the attributes hash, and also type being present in the json result.

```
env.serializer.serialize(snapshot, { includeId:true, type:type.typeKey })
```

**Comment by barelyknown** [comment]
You're right. Updated.

File: src/json-api-serializer.js [hunk 2]
```diff
@@ -181,6 +188,14 @@ DS.JsonApiSerializer = DS.RESTSerializer.extend({
 
   serialize: function(snapshot, options) {
     var data = this._super(snapshot, options);
+    data['attributes'] = {};
+    for (var key in data) {
+      if (key === 'links' || key === 'attributes' || key === 'id') {
```
Comment (eneuhauser) [inline src/json-api-serializer.js]: Type should be be ignored here as well.

**Comment by barelyknown** [comment]
Those last two commits deal with [this change in Ember Data 1.0.0-beta.17](https://github.com/emberjs/data/commit/94dfd6886b2b08019d9a6a9f940f629ed03c4efa). The snapshot's `hasMany` function can now return `undefined`, and that needs to be guarded against. I'm not sure if there are other 1.12 changes that impact this library. I haven't run into any others yet, but I'm just now working through the upgrade.

**Comment by barelyknown** [comment]
I submitted a separate PR https://github.com/eneuhauser/ember-json-api/pull/14 to master that includes changes for JSON API 1.0.

**Pull request merged**